File: app/src/main/java/com/example/myapplication/graphdatabase/Node.java
package com.example.myapplication.graphdatabase;

public class Node {
    public static int currentNodeCount;
    public String activityName;
    public String clickableElements;
    public String description;
    public String screenDumperXML;
    public int nodeID;

    public Node(String activityName, String clickableElements, String description, String screenDumperXML) {
        this.activityName = activityName;
        this.clickableElements = clickableElements;
        this.description = description;
        this.screenDumperXML = screenDumperXML;
        currentNodeCount += 1;
        nodeID = Utils.generateID();
    }

    public Node( String activityName, String clickableElements, String description, String screenDumperXML, int nodeID) {
        this.activityName = activityName;
        this.clickableElements = clickableElements;
        this.description = description;
        this.screenDumperXML = screenDumperXML;
        this.nodeID = nodeID;
    }

    @Override
    public String toString(){
        StringBuilder sb = new StringBuilder();
        sb.append("nodeID:")
                .append(nodeID)
                .append("ActivityName:")
                .append(activityName)
                .append("Description:")
                .append(description);
        return sb.toString();
    }
}

File: py_version/graph/node.py
from .utils import Utils
class Node:
    current_node_count = 0

    def __init__(self, activity_name, clickable_elements, description, screen_dumper_xml, node_id=None):
        self.activity_name = activity_name
        self.clickable_elements = clickable_elements
        self.description = description
        self.screen_dumper_xml = screen_dumper_xml
        if node_id is None:
            Node.current_node_count += 1
            self.node_id = Utils.generate_id()
        else:
            self.node_id = node_id

    def __str__(self):
        return f"nodeID:{self.node_id} ActivityName:{self.activity_name} Description:{self.description}"
